packages/store/src/symbols.ts

```diff
@@ -63,3 +63,9 @@ export interface ActionOptions {
    */
   cancelUncompleted: boolean;
 }
+
+export interface NgxsOnInit {
+  ngxsOnInit(ctx?: StateContext<any>): void | any;
+}
+
+export type NgxsLifeCycle = Partial<NgxsOnInit>;
```

Comment:
this makes sure that we know internally that lifecycle methods are optional. we can also add additional life cycle interfaces like this:

```TS
export type NgxsLifeCycle = Partial<NgxsOnInit, NgxsOnDestroy>;
```

Reply:
when would destroy ever happen?

Reply:
oh i'm not saying it would. I am just showing an example of how we can compose the lifecycle in a typesafe way